openshift/openshift-ansible-contrib
# Support Cinder-backed Openshift registry

#### What does this PR do?
It lets you set up the openshift registry with an existing pre-configured Cinder volume.


#### How should this be manually tested?
1. Before provisioning run: `openstack volume create --size 10 openshift-registry` and note it's UUID.
2. Add `prepare_and_format_registry_volume: true` to `inventory/group_vars/all.yml`
3. Add your openstack credentials to `inventory/group_vars/OSEv3.yml`

```
openshift_cloudprovider_kind: openstack
openshift_cloudprovider_openstack_username: <username>
openshift_cloudprovider_openstack_password: <password>
openshift_cloudprovider_openstack_auth_url: <auth url>
openshift_cloudprovider_openstack_tenant_name: <tenant name>
```
4. Configure openshift-ansible to use the Cinder volume by adding this to `inventory/group_vars/OSEv3.yml`:

```
openshift_hosted_registry_storage_kind: openstack
openshift_hosted_registry_storage_access_modes: ['ReadWriteOnce']
openshift_hosted_registry_storage_openstack_filesystem: xfs
openshift_hosted_registry_storage_openstack_volumeID: <volume uuid>
openshift_hosted_registry_storage_volume_size: 10Gi
```

5. Run the end to end tests -- they should all pass.
6. Validate that the registry is attached to the infra node: `openstack volume list` -- the volume should be `in-use`, attached to the infra node usually as `/dev/vdb`
7. SSH to the infra node and verify that the volume is mounted: `mount | grep vdb` (this should show a couple of mount points)

#### Is there a relevant Issue open for this?
n/a

#### Who would you like to review this?
cc: @tzumainn @Tlacenka  @bogdando PTAL
